Classes/common/CDTReplicator/CDTAbstractReplication.m

```diff
@@ -19,6 +19,14 @@
 
 NSString *const CDTReplicationErrorDomain = @"CDTReplicationErrorDomain";
 
+@interface CDTAbstractReplication ()
+
+NS_ASSUME_NONNULL_BEGIN
+@property (nonnull, nonatomic, readwrite, strong) NSArray *httpInterceptors;
```

Comment:
Why not just use the `addInterceptors:` method rather than making this readwrite, looks like you just use this in `copy`?